src/service/viewer-impl.js

```diff
@@ -766,16 +759,15 @@ export class Viewer {
     if (eventType == 'viewport') {
       if (data['paddingTop'] !== undefined) {
         this.paddingTop_ = data['paddingTop'];
-        this.viewportObservable_.fire(
-          /** @type {!JSONType} */ (data));
+        this.messageObservables_[eventType].fire(
+            /** @type {!JSONType} */ (data));
         return Promise.resolve();
       }
       return undefined;
     }
     if (eventType == 'historyPopped') {
-      this.historyPoppedObservable_.fire({
-        newStackIndex: data['newStackIndex'],
-      });
+      this.messageObservables_[eventType].fire(
```

Comment:
This API will grow because our messaging is a two-way protocol. In other words, it's very likely that some message handlers may be asked to respond back. Since our observables do not support any response type, we can instead utilize `ViewerMessage` and add methods such as `ViewerMessage.respond(...)`.